backend/models/userModel.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+
+const userSchema = mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    email: {
+      type: String,
+      required: true,
+      unique: true,
+    },
+    password: {
+      type: String,
+      required: true,
+    },
+  },
+  {
+    // tells mongoose to assign createdAt and updatedAt fields to the user schema
+    timestamps: true,
+  }
+);
+
+userSchema.methods.matchPassword = async function (enteredPassword) {
```

Comment:
If I'm not mistaken, I don't think the matchPassword method needs to be `async` nor would we need `await` since a boolean should only be returned rather than a promise

Reply:
Fixed: I think you are correct and the changes are reflected in recent commits